src/lib.rs

```diff
@@ -0,0 +1,222 @@
+use pyo3::ffi::{
+    PyContextVar_Get, PyEval_SetProfile, PyFrameObject, PyUnicode_AsUTF8, Py_tracefunc,
+    CO_ASYNC_GENERATOR, CO_COROUTINE, CO_ITERABLE_COROUTINE,
+};
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+use pyo3::{AsPyPointer, PyAny, Python};
+use std::collections::HashMap;
+use std::convert::{TryFrom, TryInto};
+use std::ffi::CStr;
+use std::os::raw::c_int;
+use std::time::Instant;
+
+/// Enum of possible Python's Trace/Profiling events
+#[allow(dead_code)]
+#[repr(i32)]
+enum TraceEvent {
+    Call,
+    Exception,
+    Line,
+    Return,
+    CCall,
+    CException,
+    CReturn,
+    Opcode,
+    __NonExhaustive,
+}
+
+impl TryFrom<c_int> for TraceEvent {
+    type Error = &'static str;
+    /// Cast i32 event (raw from Python) to Rust enum.
+    fn try_from(value: i32) -> Result<Self, Self::Error> {
+        if value > 7 || value < 0 {
+            return Err("Not valid enum value");
+        }
+        Ok(unsafe { std::mem::transmute(value) })
```

Comment:
Don't hardcode the number of `TraceEvent` variants, instead maybe have
```suggestion
const LARGER_THAN_ANY_TRACE_EVENT: i32 = TraceEvent::Opcode as i32 + 1;
...
        if value >= LARGER_THAN_ANY_TRACE_EVENT {...}
```

Another approach would be matching `value` against every variant's numeric value. That way you don't need the `unsafe` call to `transmute` and because `TraceEvent` is non-exhaustive whenever you add a new variant it would have to matched to get the code to compile, which is also good. You might pay a little in performance, though.